### PR TITLE
[WFCORE-3078]: Intermittent failure in AuditLogBootingSyslogTest shows that WFCORE-2923 doesn't cover every case

### PR DESCRIPTION
**Pull request by ehsavoie**
Changing the AuditLogger in the ResultHandler of AuditLogLoggerAddHandler instead of as a RUNTIME step.
Clearing the queue when writting log entries after the ManagedAuditLogger change status to LOGGING.

Jira: https://issues.jboss.org/browse/WFCORE-3078
JBEAP: https://issues.jboss.org/browse/JBEAP-11387

### CODE REVIEW

File: domain-management/src/main/java/org/jboss/as/domain/management/audit/AuditLogLoggerResourceDefinition.java [hunk 1]
```diff
@@ -153,25 +154,11 @@ public void execute(final OperationContext context, final ModelNode operation) t
...
-                                });
-                            }
-                        }, OperationContext.Stage.RUNTIME);
-
-
-                        context.completeStep(new OperationContext.RollbackHandler() {
-                            @Override
-                            public void handleRollback(OperationContext context, ModelNode operation) {
+                            } else {
+                                auditLoggerProvider.setLoggerStatus(oldStatus);
```
Comment (bstansberry) [inline domain-management/src/main/java/org/jboss/as/domain/management/audit/AuditLogLoggerResourceDefinition.java]: You shouldn't need this since if this else block is entered this OSH is not calling setLoggerStatus anywhere else.

File: testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/auditlog/AuditLogBootingSyslogTest.java [hunk 2]
```diff
@@ -115,8 +115,7 @@ public void testSyslog() throws Exception {
         final BlockingQueue<SyslogServerEventIF> queue = BlockedSyslogServerEventHandler.getQueue();
         queue.clear();
         container.start();
-        long endTime = System.currentTimeMillis() + TimeoutUtil.adjust(5000);
-        waitForExpectedQueueSize(18, queue);
+        waitForExpectedQueueSize(1, queue);
```
Comment (bstansberry) [inline testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/auditlog/AuditLogBootingSyslogTest.java]: Why this change? We shouldn't be changing the number of logged items.

Comment (ehsavoie) [inline testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/auditlog/AuditLogBootingSyslogTest.java]: This is due to the fact that there was a bug in boot : we didn't clear the queue when writing log entries on status change. The 18 was due to leftovers from the previous operations (we didn't understand that at that  time). Now this is coherent with the AuditLogBootingLogTestCase.

**Comment by ehsavoie** [comment]
retest this please

**Comment by bstansberry** [comment]
retest this please

File: controller/src/main/java/org/jboss/as/controller/audit/ManagedAuditLoggerImpl.java [hunk 3]
```diff
@@ -226,6 +226,7 @@ public void setLoggerStatus(final Status newStatus) {
                     } catch (Exception e) {
                         handleLoggingException(e);
                     }
+                    queuedItems.clear();
```
Comment (rsvoboda) [inline controller/src/main/java/org/jboss/as/controller/audit/ManagedAuditLoggerImpl.java]: `queuedItems.clear()` inside `for (AuditLogItem record : queuedItems)`
so only one AuditLogItem is written ?

Comment (ehsavoie) [inline controller/src/main/java/org/jboss/as/controller/audit/ManagedAuditLoggerImpl.java]: Correct :( this is a bug

Comment (mchoma) [inline controller/src/main/java/org/jboss/as/controller/audit/ManagedAuditLoggerImpl.java]: tracked by WFCORE-3109
